lambdas/revoke-vcs/src/main/java/uk/gov/di/ipv/core/revokevcs/RevokeVcsHandler.java

```diff
@@ -0,0 +1,108 @@
+import uk.gov.di.ipv.core.revokevcs.domain.RevokeRequest;
+import uk.gov.di.ipv.core.revokevcs.domain.UserIdCriIdPair;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+@ExcludeFromGeneratedCoverageReport
+@SuppressWarnings("unused") // Temporarily disable to pass sonarqube
+public class RevokeVcsHandler implements RequestStreamHandler {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final ConfigService configService;
+    private final VerifiableCredentialService verifiableCredentialService;
+    private final DataStore<VcStoreItem> dataStore;
+
+    @SuppressWarnings("unused") // Used by AWS
+    public RevokeVcsHandler(
+            ConfigService configService,
+            VerifiableCredentialService verifiableCredentialService,
+            DataStore<VcStoreItem> dataStore) {
+        this.configService = configService;
+        this.verifiableCredentialService = verifiableCredentialService;
+        this.dataStore = dataStore;
+    }
+
+    @SuppressWarnings("unused") // Used through dependency injection
+    public RevokeVcsHandler() {
+        this.configService = new ConfigService();
+        this.verifiableCredentialService = new VerifiableCredentialService(configService);
+        boolean isRunningLocally = this.configService.isRunningLocally();
+        this.dataStore =
+                new DataStore<>(
+                        this.configService.getEnvironmentVariable(
+                                EnvironmentVariable.REVOKED_USER_CREDENTIALS_TABLE_NAME),
+                        VcStoreItem.class,
+                        DataStore.getClient(isRunningLocally),
+                        isRunningLocally,
+                        configService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context)
+            throws IOException {
+        LogHelper.attachComponentIdToLogs(configService);
+        List<UserIdCriIdPair> userIdCriIdPairs =
+                new ObjectMapper()
+                        .readValue(inputStream, RevokeRequest.class)
+                        .getUserIdCriIdPairs();
+        LOGGER.info("Revoking {} VCs", userIdCriIdPairs.size());
+
+        revoke(userIdCriIdPairs);
+        LOGGER.info("Finished attempt to revoke {} VCs", userIdCriIdPairs.size());
+    }
+
+    private void revoke(List<UserIdCriIdPair> userIdCriIdPairs) {
+        for (int i = 0; i < userIdCriIdPairs.size(); i++) {
+            var userIdCriIdPair = userIdCriIdPairs.get(i);
+            LOGGER.info("Processing VC {} / {}", i, userIdCriIdPairs.size());
+
+            try {
+                revoke(userIdCriIdPair);
+                LOGGER.info("Successfully revoked VC.");
+            } catch (Exception e) {
+                LOGGER.error("Failed to revoke VC.");
```

Comment:
Can we log the exception as well? This won't give us any information to debug why it failed

Reply:
Hmm, that's true, but I think it's reasonable to see unexpected details otherwise we have no ability to debug. If there are particular cases you're worried about, then we could catch them explicitly.